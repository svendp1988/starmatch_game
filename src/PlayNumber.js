import React from "react";

export class PlayNumber extends React.Component {
    render() {
        return (
            <button
                className="number"
                onClick={() => this.props.onClick(this.props.number, this.props.status)}
                style={{backgroundColor: colors[this.props.status]}}
            >
                {this.props.number}
            </button>
        )
    }

}

// Color Theme
const colors = {
    available: 'lightgray',
    used: 'lightgreen',
    wrong: 'lightcoral',
    candidate: 'deepskyblue',
};
