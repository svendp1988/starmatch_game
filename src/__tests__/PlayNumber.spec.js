import React from 'react';
import {shallow} from 'enzyme';
import PlayNumber from "../PlayNumber";
import renderer from 'react-test-renderer';


describe('PlayNumber', function () {
    it('should match snapshot', function () {
        let wrapper = renderer.create(
            <PlayNumber/>
        ).toJSON();
        expect(wrapper).toMatchSnapshot();
    });

    it('should have a lightgray background color if it is available and display the correct number as text', () => {
        const props = {
            number: 14,
            status: "available"
        }
        const wrapper = shallow(<PlayNumber {...props}/>);

        const button = wrapper.get(0);
        expect(button).toHaveProperty("type", "button");
        expect(button.props.style).toEqual({ backgroundColor: 'lightgray' });
        expect(wrapper.text()).toBe("14");
    })
});
