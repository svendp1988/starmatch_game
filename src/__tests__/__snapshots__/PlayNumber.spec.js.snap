// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayNumber should match snapshot 1`] = `
<button
  className="number"
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": undefined,
    }
  }
/>
`;
